packages/braid-design-system/src/lib/components/private/Popover/Popover.tsx

```diff
@@ -0,0 +1,342 @@
+import { assignInlineVars } from '@vanilla-extract/dynamic';
+import dedent from 'dedent';
+import {
+  type ReactNode,
+  useEffect,
+  useState,
+  forwardRef,
+  useRef,
+  useImperativeHandle,
+  useCallback,
+  type RefObject,
+} from 'react';
+
+import type { ResponsiveSpace } from '../../../css/atoms/atoms';
+import { useIsomorphicLayoutEffect } from '../../../hooks/useIsomorphicLayoutEffect';
+import { Box } from '../../Box/Box';
+import { BraidPortal } from '../../BraidPortal/BraidPortal';
+
+import * as styles from './Popover.css';
+
+const animationDelayInMs = 250;
+
+type Placement = 'top' | 'bottom';
+
+export interface PopoverProps {
+  id?: string;
+  role?: string;
+  align?: 'left' | 'right' | 'center';
+  width?: 'content' | 'full';
+  placement?: Placement;
+  lockPlacement?: boolean;
+  offsetSpace?: ResponsiveSpace;
+  open: boolean;
+  onClose?: () => void;
+  triggerRef: RefObject<any>;
+  enterFocusRef?: RefObject<any>;
```

Comment:
Should be able to use the base `HTMLElement` for these:

```suggestion
  triggerRef: RefObject<HTMLElement>;
  enterFocusRef?: RefObject<HTMLElement>;
```